src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace_XmlConsumption.cs

```diff
@@ -445,6 +446,18 @@ private static string GetLanguage(TestWorkspace workspace, XElement projectEleme
             return languageName;
         }
 
+        private static string GetDefaultNamespace(TestWorkspace workspace, XElement projectElement)
+        {
+            var language = GetLanguage(workspace, projectElement);
+            if (language != LanguageNames.CSharp)
+            {
+                return null;
```

Comment:
why?  what about VB?  Cn't they specify their root namespace as well?

Reply:
Yes and no. 

While both csproj and vbproj use `<rootnamespace>` property, they are very different things (See [this article](http://spiderinnet1.typepad.com/blog/2011/12/vbnet-root-namespace-vs-c-default-namespace.html) for more details). 

In C#, it's called default namespace (even though we got the value from `rootnamespace` property), and it doesn't affect the semantic of the code in anyway, pretty much just something used by VS. For example, when you create a new class, the namespace for the new class is based on it. Therefore, we can't get this info from compiler.

However, in VB, it's called root namespace, and that info is part of the VB compilation (parsed from arguments), because VB compiler need it to determine the root of all the namespace declared in the compilation. as a result of this, this whole "default namespace must be containing namespace of declared namespace" check I'm doing currently is irrelevant to VB, the VB compiler makes sure of that.

Right now the logic in the abstract provider are developed based C#, many of the checks probably don't make sense for VB, I just haven't give it much thought yet. Will change them when I started working on teh VB side.